### ISSUE TITLE
Self-hosted package installation fail

### ISSUE THREAD

**Issue by mrtaracqk** [supertokens/supertokens-core#906]
## 🐛 Bug Report

 I'm having trouble installing SuperTokens. I'm trying to install the self-hosted version on my MacBook with an M2 chip. I run ".install" and see the download process, but it crashes with the following log. 

I think this may be related to the Apple Silicon because I previously installed it successfully on my Intel MacBook.

## Useful informations

Installation log:
```shell
Downloading (33/36): https://repo1.maven.org/maven2/de/mkammerer/argon2-jvm/2.11/argon2-jvm-2.11.jar into cli
Downloading (34/36): https://repo1.maven.org/maven2/de/mkammerer/argon2-jvm-nolibs/2.11/argon2-jvm-nolibs-2.11.jar into cli
Downloading (35/36): https://repo1.maven.org/maven2/org/mindrot/jbcrypt/0.4/jbcrypt-0.4.jar into cli
Downloading (36/36): https://repo1.maven.org/maven2/net/java/dev/jna/jna/5.8.0/jna-5.8.0.jar into cli
java.io.IOException: No such file or directory
	at java.base/java.io.UnixFileSystem.createFileExclusively(Native Method)
	at java.base/java.io.File.createNewFile(File.java:1026)
	at io.supertokens.cli.commandHandler.install.InstallHandler.createSupertokensScript(InstallHandler.java:146)
	at io.supertokens.cli.commandHandler.install.InstallHandler.doCommand(InstallHandler.java:61)
	at io.supertokens.cli.commandHandler.CommandHandler.handleCommand(CommandHandler.java:31)
	at io.supertokens.cli.Main.start(Main.java:101)
	at io.supertokens.cli.Main.main(Main.java:50)

error while installing SuperTokens. Please try again
``` 

Info from version.yaml:

```
core_version: 7.0.16
plugin_interface_version: 4.0.5
plugin_version: 5.0.5
plugin_name: mysql
``` 

Setup:
M2 Pro chip, macOS Sonoma 14.2.1 (23C71)

**Comment by rishabhpoddar** [comment]
> I think this may be related to the Apple Silicon because I previously installed it successfully on my Intel MacBook.

Thats probably correct. We can investigate when we have time. In the meantime, you could try using our docker image instead.

**Comment by nyko28** [comment]
> TLDR : before running the installation script, run
> ```sh
> sudo mkdir /usr/local/bin
> ```


Hey,

I encountered the same issue on mac M3. The interesting part of the log is :
`io.supertokens.cli.commandHandler.install.InstallHandler.createSupertokensScript(InstallHandler.java:146)`

The `createSupertokensScript` method calls `getSupertokensScriptLocation` to know where to create the supertokens script file depending on the running OS. For mac we have [this condition](https://github.com/supertokens/supertokens-core/blob/691d59d383e1a93bd33f668259bb21dfbe92ca46/cli/src/main/java/io/supertokens/cli/commandHandler/install/InstallHandler.java#L227C11-L229C11) :

```java
else if (OperatingSystem.getOS() == OperatingSystem.OS.MAC) {
    return "/usr/local/bin/supertokens";
} 
```

We try then create the supertokens script under /usr/local/bin. But it appears that after a fresh install of MacOS, this "bin" directory does not necessary exists. So the you just have to create it before running the installation script and everything will be ok :)
